storage/storage.go

```diff
@@ -169,6 +169,12 @@ type Claims struct {
 	Groups []string
 }
 
+// Data needed for PKCE (RFC 7636)
+type CodeChallenge struct {
```

Comment:
I am wondering if the better names would be 
CodeChallenge -> Challenge
CodeChallengeMethod -> Method
so that we don't have excessive verbosity when used?: 
CodeChallenge.CodeChallengeMethod

alternatively the type could be renamed to PKCE

Reply:
`CodeChallenge.Challenge` also sounds verbose, so
I prefer `PKCE.CodeChallenge` and `PKCE.CodeChallengeMethod`
It makes it clear, for what it is used.